### PR TITLE
tar.gz build

### LINKED ISSUE #80: .tar.gz packaging
In addition to `.deb` and `.rpm,` we need to build `.tar.gz` packages.

### PR DESCRIPTION
**Pull request by longshorej**
Fixes #80 

This implements the build for .tar.gz by creating a self-extracting executable named `rp` as the only artifact inside the final tar.fz file.

- [x] Explore possible implementations
- [x] Solve dynamic linker issue (~probably different tar.gz for different libc, TBD~ dynamic linkers are static binaries, so simply have to execute them and interpret the ELF)
- [x] Bintray API upload

### CODE REVIEW

File: project/BuildTarget.scala [hunk 1]
```diff
@@ -205,3 +218,72 @@ case class DebBuildTarget(distributions: Seq[String], components: String, depend
...
+          |
+          |SCRIPT_NAME="$${BASH_SOURCE[0]}"
+          |
+          |if [ -h "$$SCRIPT_NAME" ]; then
+          |  SCRIPT_NAME="$$(readlink "$$SCRIPT_NAME")"
+          |fi
+          |
+          |DIR="$$(mktemp -d)"
+          |
+          |#trap 'rm -r $$DIR' EXIT
```
Comment (longshorej) [inline project/BuildTarget.scala]: TODO: uncomment this

File: project/BuildInfo.scala [hunk 2]
```diff
@@ -6,6 +6,8 @@ import AdditionalIO._
 object BuildInfo {
   val Builds =
     Seq(
+      MuslBuild.tgz("tgz"),
```
Comment (longshorej) [inline project/BuildInfo.scala]: I doubt we can actually use the MuslBuild here.

Comment (longshorej) [reply project/BuildInfo.scala]: Eh, we probably can if we rewrite part of the binary. I'll look into it.

File: project/BuildTarget.scala [hunk 3]
```diff
@@ -205,3 +220,75 @@ case class DebBuildTarget(distributions: Seq[String], components: String, depend
     AdditionalIO.setExecutable(stage / "command")
   }
 }
+
+case class TarGzSelfContainedExecutableBuildTarget(libs: Seq[String]) extends BuildTarget {
+  val architecture = "amd64"
+
+  override def launcherExecHook: String = "exec $DIR/lib/ld-musl-x86_64.so.1"
```
Comment (longshorej) [inline project/BuildTarget.scala]: The dynamic linker, `ld-musl-x86_64.so.1`, is also a static binary so we can simply exec it followed by our binary (`rp`), without having to change the hard-coded path to the linker within the binary itself.

Comment (mitkus) [reply project/BuildTarget.scala]: Neat!

**Pull request merged**